Prototype of files in project (read description)

sartfromua opened the pull request:
Сделал так. Вопрос: как правильно прочитать из файла и разделить на разные переменные несколько строк неизвестной длинны?
Пример: записал в файл 3 строки (person.name, person.surname, person.sex) как мне их отдельно в разные строки записать (длину не знаю), после них идут данные типа int.
Правильный ли include других структур в sourse файлах для описания функций?
Еще Codeblocks в подсказках почему-то выдает по 2 варианта каждой из print функций, хотя input функций по одной

sartfromua committed:
Prototype of files in project (read description) 

Сделал так. Вопрос: как правильно прочитать из файла и разделить на разные переменные несколько строк неизвестной длинны?
Пример: записал в файл 3 строки (person.name, person.surname, person.sex) как мне их отдельно в разные строки записать (длину не знаю), после них идут данные типа int.

mecchmatProjects commented on Project_27/Project_27.cbp:
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+<CodeBlocks_project_file>
+	<FileVersion major="1" minor="6" />
+	<Project>
```
please delete and do not commit oher except .c/cpp, *.h and Makefile files

sartfromua replied:
I thought that, this is my Makefile file (project file in Codeblocks)

mecchmatProjects commented on Project_27/Project_27.depend:
```diff
@@ -0,0 +1,94 @@
+# depslib dependency file v1.0
```
delete it

mecchmatProjects reviewed (changes requested):
Delete extra files. To read from text files on C use fprintf("%s" ....) function.  (Or read it char by char using getchar()...)
For binary files use fread(...)
Code looks not bad now. Only it should work. Please add some simple tests U'd more quickly found then Your bugs...

sartfromua committed:
Delete Project_27.cbp

sartfromua committed:
Delete Project_27.depend

sartfromua committed:
Delete Project_27.layout

sartfromua committed:
Delete Project_27/bin/Debug directory

sartfromua committed:
Delete Project_27/obj/Debug directory

sartfromua committed:
Delete date_test

sartfromua committed:
Made all fin fout functions and small test.

Дописал функции, вместо бинарных файлов записал все в текстовые. Проверил, все работают.
Файл Project_27.cbp у меня работает как makefile, насколько я понимаю, остальные файлы не загружал.

mecchmatProjects commented on Project_27/Sources/Person.c:
```diff
@@ -0,0 +1,64 @@
+
+
+int finPerson(const char* fname, Person person){
+    remove(fname); // not so good idea, but may work to clear file
```
wb flag could remove it by default

mecchmatProjects commented on Project_27/Sources/Person.c:
```diff
@@ -0,0 +1,64 @@
+    FILE *file = fopen(fname, "rb");
+    Person person;
+
+    // ??? how to execute only one char list, when there are 3 of them in file?
```
It can be helpful if You describe Your file structure. You'd better write somthing about it in readme.txt file and push example of it. Then myabe I can help

mecchmatProjects replied:
Also object(o) and exe files better be deleted in github

mecchmatProjects replied:
You can create Your file questions with example in separate pull request, so we can resolve this purticular one ad go to the next issue

mecchmatProjects reviewed (commented):
Fix particular my last comments (delete extra files) and we could resolve this pull request. File issue will be the next PR.

mecchmatProjects commented on Project_27/Headers/Product.h:
```diff
@@ -0,0 +1,16 @@
+#include "Equipment.h"
+
+typedef struct {
+    char name[255];
```
1 option) - read/write 255 chars
2 option - use text files and read fscanf("%s")
3 option (best) - add:
 size_t len, 
char * name;

then read len,  read next len chars....

sartfromua commented:
You checked outdated files, i don't know how, bcs i updated them yesterday and you rewiewed them today, all issues were fixed by used txt files to save instead of bin.
